command/hcl2_upgrade.go

```diff
@@ -282,6 +283,52 @@ func fallbackReturn(err error, s []byte) []byte {
 	return append([]byte(fmt.Sprintf("\n# could not parse template for following block: %q\n", err)), s...)
 }
 
+// reTemplate replaces all instances of plain accesses to variables by their
+// escaped version so that go template does not attempt to execute them
+func reTemplate(nd parse.Node, out io.Writer, funcs texttemplate.FuncMap) error {
+	switch node := nd.(type) {
+	case *parse.ActionNode:
+		// Leave pipes as-is
+		if len(node.Pipe.Cmds) > 1 {
+			fmt.Fprintf(out, "%s", node.String())
+			return nil
+		}
+
+		cmd := node.Pipe.Cmds[0]
+		args := cmd.Args
+		if len(args) > 1 {
+			// Function calls with parameters are left aside
+			fmt.Fprintf(out, "%s", node.String())
+		} else {
+			_, ok := funcs[args[0].String()]
+			if ok {
+				// Known functions left as-is
+				fmt.Fprintf(out, "%s", node.String())
+			} else {
+				// Escape anything that isn't in the func map
+				fmt.Fprintf(out, "{{ \"{{\" }} %s {{ \"}}\" }}", cmd.String())
+			}
+		}
```

Comment:
nit: We can return early and drop the else
```go
		if len(args) > 1 {
			// Function calls with parameters are left aside
			fmt.Fprintf(out, "%s", node.String())
			return nil
		} 
		_, ok := funcs[args[0].String()]
		if ok {
			// Known functions left as-is
			fmt.Fprintf(out, "%s", node.String())
                        return nil
		} 
		// Escape anything that isn't in the func map
		fmt.Fprintf(out, "{{ \"{{\" }} %s {{ \"}}\" }}", cmd.String())		
	}
```

Reply:
I like linearizing code, if the behaviour is the same, I'll definitely consider doing that, thanks for the suggestion!